.travis.yml

```diff
@@ -21,20 +27,24 @@ jobs:
     - docker run --env-file .env.test my-test-image test_items.py
     - docker run -e Mongo_Url my-test-image test_system.py
 
-  - stage: dockerhub
-    script: skip
-    deploy:
-      script:
-      - echo $DockerPassword | docker login --username joefish29 --password-stdin
-      - docker build --target production --tag joefish29/todo-app:latest .
-      - docker push joefish29/todo-app:latest
-    on:
-      branch: master
-
   - stage: deploy
-    script: skip
+    install:
+    - wget https://releases.hashicorp.com/terraform/"$TF_VERSION"/terraform_"$TF_VERSION"_linux_amd64.zip
+    - unzip terraform_"$TF_VERSION"_linux_amd64.zip
+    - sudo mv terraform /usr/local/bin/
+    - rm terraform_"$TF_VERSION"_linux_amd64.zip
+    - terraform init
+
+    before_deploy:
+    - terraform apply -var prefix=todoapp -var location=uksouth -var clientId=clientId -var client_secret=client_secret -auto-approve
```

Comment:
Kind of misleading to all of these steps in before_deploy

Also, it would be nice to run terraform plan to log the details of the changes.